modules/internal/ChapelGpuSupport.chpl

```diff
@@ -23,16 +23,19 @@ module ChapelGpuSupport {
   use ChplConfig;
 
   extern var chpl_gpu_debug : bool;
-
   config const debugGpu = false;
 
+  extern var chpl_gpu_no_cpu_mode_warning: bool;
+  config const gpuNoCpuModeWarning = isEnvSet("CHPL_GPU_NO_CPU_MODE_WARNING");
```

Comment:
There's something about `CHPL_GPU_NO_CPU_MODE_WARNING` that makes it seem different from other `CHPL_` variables:

* It doesn't get printed out by `printchplenv --all`.
* I don't think it would get addressed if put in a Chapel configuration file (https://chapel-lang.org/docs/usingchapel/chplenv.html#chapel-configuration-files).
* It's strictly a runtime configuration thing that doesn't impact the build of either the compiler or the runtime library.

I suppose `CHPL_DEBUG` doesn't fit these first two bullets. We also have `CHPL_RT_` variables, is this more like one of those?

Do we have a precedent for other `CHPL_` variables that impact runtime behavior / are used to assign values to config consts like this?

I'm not opposed to having this as a `CHPL_` variable but this seems different enough from what we do with other `CHPL_` variables I'm familiar with that I question if this is somehow falling out of our overall design philosophy for these. If we have a precedent of doing this ignore my comment but otherwise you might want to get a second opinion.

Reply:
> - It doesn't get printed out by printchplenv --all
> - I don't think it would get addressed if put in a Chapel configuration file (https://chapel-lang.org/docs/usingchapel/chplenv.html#chapel-configuration-files).

Not all environment variables (either `CHPL_` or `CHPL_RT_`) we care about do. Note that for most (all?) compiler flags, there's a corresponding environment variable that we take a look, as well. So, we don't try to be exhaustive with respect to those bullets.

> Do we have a precedent for other CHPL_ variables that impact runtime behavior / are used to assign values to config consts like this?

Maybe not `config const`, but here's an example that I am aware and based this off of:
https://github.com/chapel-lang/chapel/blob/main/modules/packages/CopyAggregation.chpl#L95-L97

> I'm not opposed to having this as a CHPL_ variable but this seems different enough from what we do with other CHPL_ variables I'm familiar with that I question if this is somehow falling out of our overall design philosophy for these. If we have a precedent of doing this ignore my comment but otherwise you might want to get a second opinion.

That's a valid question, but based on my responses above; I am not bothered.

---

OTOH,

> We also have CHPL_RT_ variables, is this more like one of those?

This is true. Maybe `CHPL_RT_` is a more consistent prefix for this. If you prefer that, I am happy to make that change.

Reply:
> Not all environment variables (either CHPL_ or CHPL_RT_) we care about do. Note that for most (all?) compiler flags, there's a corresponding environment variable that we take a look, as well. So, we don't try to be exhaustive with respect to those bullets.

Ok cool. I just wanted to check that we weren't going "out of model" with this. I'd be fine having with either as a `CHPL_` or `CHPL_RT_` variable then.

Reply:
Leaving as-is.